src/quic/node_quic_session.cc

```diff
@@ -393,19 +393,28 @@ void JSQuicSessionListener::OnClientHello(
   HandleScope scope(env->isolate());
   Context::Scope context_scope(env->context());
 
+  Local<Array> ciphers;
+  Local<Value> alpn_string = Undefined(env->isolate());
+  Local<Value> servername = Undefined(env->isolate());
+
+  // TODO(@jasnell): Need to decide how to handle the possible
+  // ToLocal failures more gracefully than crashing.
+
+  CHECK(session()->crypto_context()->hello_ciphers().ToLocal(&ciphers));
+
+  if (alpn != nullptr)
+    CHECK(String::NewFromUtf8(env->isolate(), alpn).ToLocal(&alpn_string));
+
+  if (server_name != nullptr)
+    CHECK(String::NewFromUtf8(env->isolate(), server_name)
+        .ToLocal(&servername));
```

Comment:
`CHECK(x.ToLocal(&y))` might as well be `y = x.ToLocalChecked()`. That being said, it would be really really nice to not crash, and I don’t see any reason why we can’t return here if there’s an exception instead..

Reply:
Yeah... Was still thinking through the best way to handle those. I'm thinking that it definitely should not crash but it should be fatal for the QuicSession.

Reply:
That sounds good. We'll still want to destroy the QuicSession tho...